### ISSUE TITLE
Game gets stuck on Exiting when Limit is reached

### ISSUE THREAD

**Issue by troyBORG** [dfgHiatus/CacheGetClapped#5]
So now that my Cache has reached its size limit I set.  I noticed now when I quit the game even if I leave it running overnight on the exiting screen it will still never close.

I know its this mod, as I removed it and now the game exits right away.

I didn't have this issue in Neos.  So there must be something that isn't working as intended?

I did notice this seems to be in a few log...
```
Exception:
System.IO.IOException: The process cannot access the file 'D:\Resonite_Cache\Cache\7rsmc1c5s6uabrgh' because it is being used by another process.
  at System.IO.__Error.WinIOError (System.Int32 errorCode, System.String maybeFullPath) [0x001d7] in <9577ac7a62ef43179789031239ba8798>:0 
  at System.IO.FileInfo.Delete () [0x00038] in <9577ac7a62e
```

**Comment by dfgHiatus** [comment]
Thanks for bringing this to my attention. I've added some logic to handle this, and I've pushed the source for this too: [CacheGetClapped.zip](https://github.com/dfgHiatus/CacheGetClapped/files/13299199/CacheGetClapped.zip)

**Comment by troyBORG** [comment]
I forgot to mention.  I had mine set to 60 days or 50GB.

I just tried with your new version (of coarse last night like an idiot I did manually delete some of the files to bring the cache down)
But I did go in an set the limit to something like 40GB and I did see the game exits correctly, and when I checked the cache folder it did bring the size down.  So it appears to be working now.

I guess if I run into any other issues I'll let you know.
